scripts/check_frontend_coverage_test.py

```diff
@@ -0,0 +1,335 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+import os
+import shutil
+import subprocess
+from core.tests import test_utils
+import python_utils
+
+from . import check_frontend_coverage
+
+
+class CheckFrontEndCoverageTests(test_utils.GenericTestBase):
+
+    def setUp(self):
+        super(CheckFrontEndCoverageTests, self).setUp()
+        self.lcov_items_list = None
+        self.check_function_calls = {
+            'open_file_is_called': False,
+            'exists_is_called': False,
+            'check_call_is_called': False,
+        }
+        self.expected_check_function_calls = {
+            'open_file_is_called': True,
+            'exists_is_called': True,
+            'check_call_is_called': True,
+        }
+        self.check_function_calls_counter = {
+            'mock_print': 0
+        }
+        self.print_messages = []
+        class MockFile(python_utils.OBJECT):
+            def __init__(self, lcov_items_list):
+                self.lcov_items_list = lcov_items_list
+            def read(self):
+                return self.lcov_items_list
+        def mock_open_file(file, option):
+            self.check_function_calls['open_file_is_called'] = True
+            return MockFile(self.lcov_items_list)
+        def mock_exists(unused_path):
+            self.check_function_calls['exists_is_called'] = True
+            return True
+        def mock_print(message):
+            self.check_function_calls_counter['mock_print'] += 1
+            self.print_messages.append(message)
+        def mock_check_call(command):
+            self.check_function_calls['check_call_is_called'] = True
+        self.open_file_swap = self.swap(python_utils, 'open_file',
+            mock_open_file)
+        self.exists_swap = self.swap(os.path, 'exists', mock_exists)
+        self.print_swap = self.swap(python_utils, 'PRINT', mock_print)
+        self.check_call_swap = self.swap(subprocess, 'check_call',
+            mock_check_call)
+        
+    def test_run_frontend_tests_script(self):
+        with self.check_call_swap:
+            check_frontend_coverage.run_frontend_tests_script()
+        self.assertEqual(self.check_function_calls['check_call_is_called'],
+            True)
+
+    def test_run_frontend_tests_script_error(self):
+        def mock_check_call(test):
+            raise subprocess.CalledProcessError(1, '')
+        
+        check_call_swap = self.swap(subprocess, 'check_call', mock_check_call)
+        with check_call_swap:
+            with self.assertRaises(SystemExit) as error:
+                check_frontend_coverage.run_frontend_tests_script()
+        self.assertEqual(error.exception.code, 1)
```

Comment:
Use a nontrivial error message and assert the error message too.

Reply:
Done.